apis/r/src/rutilities.cpp

```diff
@@ -544,3 +544,51 @@ const char *_tiledb_layout_to_string(tiledb_layout_t layout) {
         Rcpp::stop("unknown tiledb_layout_t (%d)", layout);
     }
 }
+
+// internal helper function for
+// `_get_filter_options()`
+// <Numeric> should be `int32_t` or `double`
+// TODO: replace with safer API from core when available
+template <typename Numeric>
+Numeric _get_filter_option(Rcpp::XPtr<tiledb::Filter> filter, tiledb_filter_option_t option) {
+    try {
+        switch (option) {
+        case TILEDB_BIT_WIDTH_MAX_WINDOW:
+        case TILEDB_POSITIVE_DELTA_MAX_WINDOW:
+            return static_cast<Numeric>(filter->get_option<uint32_t>(option));
+        case TILEDB_SCALE_FLOAT_BYTEWIDTH:
+            return static_cast<Numeric>(filter->get_option<uint64_t>(option));
+        default:
+            return filter->get_option<Numeric>(option);
+        }
+    } catch (const tiledb::TileDBError& e) {
+        const std::string msg = e.what();
+        if (msg.find("unknown option") != std::string::npos) {
+            if (std::is_same<Numeric, double>::value) {
+                return R_NaReal;
+            }
+            return R_NaInt;
+        }
+        if (msg.find("Filter does not support options") != std::string::npos) {
+            if (std::is_same<Numeric, double>::value) {
+                return R_NaReal;
+            }
+            return R_NaInt;
+        }
+        throw(e);
+    }
+}
+
+// filter options taken from tiledb-r
+// https://github.com/TileDB-Inc/TileDB-R/blob/525bdfc0f34aadb74a312a5d8428bd07819a8f83/src/libtiledb.cpp#L369-L388
+Rcpp::List _get_filter_options(Rcpp::XPtr<tiledb::Filter> filter) {
+    return Rcpp::List::create(
+        Rcpp::Named("filter_type") = tiledb::Filter::to_str(filter->filter_type()),
+        Rcpp::Named("compression_level") = _get_filter_option<int32_t>(filter, TILEDB_COMPRESSION_LEVEL),
+        Rcpp::Named("bit_width") = _get_filter_option<int32_t>(filter, TILEDB_BIT_WIDTH_MAX_WINDOW),
+        Rcpp::Named("positive_delta") = _get_filter_option<int32_t>(filter, TILEDB_POSITIVE_DELTA_MAX_WINDOW),
+        Rcpp::Named("float_bytewidth") = _get_filter_option<double>(filter, TILEDB_SCALE_FLOAT_BYTEWIDTH),
+        Rcpp::Named("float_factor") = _get_filter_option<double>(filter, TILEDB_SCALE_FLOAT_FACTOR),
+        Rcpp::Named("float_offset") = _get_filter_option<double>(filter, TILEDB_SCALE_FLOAT_OFFSET)
```

Comment:
Not all of these options make sense for all filters. It would be better to have a logic of `if (filter is compression filter) { add "compression_level" to the list }` and so on for the other filter types. This would avoid the `try/catch`es above.